### ISSUE TITLE
Simplify arguments of estimate_risk_spread

### ISSUE THREAD

**Issue by thibautjombart** [reconhub/epiflows#16]
The function currently has many arguments:
https://github.com/reconhub/epiflows/blob/4ccd83e9f07aa5228c47abf9858c3656aea13131/R/estimate_risk_spread.R#L42-L55

If possible, simplify the arguments, or put everything in a list which has defaults and can be checked, a la `outbreaker2`:
https://github.com/reconhub/outbreaker2/blob/master/R/create_config.R

**Comment by zkamvar** [comment]
I'm almost at the point where I can implement the method for the new epiflows class. I believe I can automate most of the parameters and leave the rest to the user:

```r
estimate_risk_spread(
  epiflows      = ef,
  location_code = "Espirito Santo",
  r_incubation  = function(n) rlnorm(n, 1.46, 0.35),
  r_infectious  = function(n) rnorm(n, 4.5, 1.5/1.96),
  n_sim         = 1e5
)
```

Some things I would need to do:

 - Create a much simpler example
 - Add the `estimate_risk_spread()` arguments to the optional variables
 - Document these variables
 - Allow users to change the variables
 - make `estimate_risk_spread()` a generic method that will default to the current implementation


Currently (as of 66367ef9d164109edbcd94ba58ae565034adaf9b), I can get it to work like so:

### Data Setup

``` r
library("epiflows")
library("epicontacts")
data(YF_Brazil)
from  <- as.data.frame.table(YF_Brazil$T_D, stringsAsFactors = FALSE)
to    <- as.data.frame.table(YF_Brazil$T_O, stringsAsFactors = FALSE)[c(2, 1, 3)]
flows <- rbind(setNames(from, c("from", "to", "n")),
               setNames(to, c("from", "to", "n")))
locations <- YF_Brazil$states
others    <- setdiff(flows$to, YF_Brazil$states$location_code)
los       <- data.frame(location_code = names(YF_Brazil$length_of_stay), 
                        length_of_stay = YF_Brazil$length_of_stay)
locations <- merge(locations,
                   data.frame(location_code = others, stringsAsFactors = FALSE),
                   by = "location_code", all = TRUE)
locations <- merge(locations, los, by = "location_code", all = TRUE)
ef <- make_epiflows(flows, 
                    locations, 
                    pop_size = "location_population",
                    duration_stay = "length_of_stay")

## Thinning the data to the location of interest, 
## though this can be done inside the model function :)
ef.es <- thin(ef[, j = "Espirito Santo", contacts = "either"])
ef.es
#> 
#> /// Epidemiological Flows //
#> 
#>   // class: epiflows, epicontacts
#>   // 11 locations; 20 flows; directed
#>   // optional variables: pop_size, duration_stay 
#> 
#>   // locations
#> 
#> # A tibble: 11 x 6
#>    id               location_populati… num_cases_time_wi… first_date_cases
#>    <chr>                         <dbl>              <dbl> <fct>           
#>  1 Espirito Santo              3973697               2600 2017-01-04      
#>  2 Italy                            NA                 NA <NA>            
#>  3 Spain                            NA                 NA <NA>            
#>  4 Portugal                         NA                 NA <NA>            
#>  5 Germany                          NA                 NA <NA>            
#>  6 United Kingdom                   NA                 NA <NA>            
#>  7 United States o…                 NA                 NA <NA>            
#>  8 Argentina                        NA                 NA <NA>            
#>  9 Chile                            NA                 NA <NA>            
#> 10 Uruguay                          NA                 NA <NA>            
#> 11 Paraguay                         NA                 NA <NA>            
#> # ... with 2 more variables: last_date_cases <fct>, length_of_stay <dbl>
#> 
#>   // flows
#> 
#> # A tibble: 20 x 3
#>    from                     to                            n
#>    <chr>                    <chr>                     <dbl>
#>  1 Espirito Santo           Italy                     2828.
#>  2 Espirito Santo           Spain                     3270.
#>  3 Espirito Santo           Portugal                  3264.
#>  4 Espirito Santo           Germany                   1898.
#>  5 Espirito Santo           United Kingdom            1985.
#>  6 Espirito Santo           United States of America 13597.
#>  7 Espirito Santo           Argentina                 5899.
#>  8 Espirito Santo           Chile                     2794.
#>  9 Espirito Santo           Uruguay                   2630.
#> 10 Espirito Santo           Paraguay                  1164.
#> 11 Italy                    Espirito Santo            1566.
#> 12 Spain                    Espirito Santo            1171.
#> 13 Portugal                 Espirito Santo            1258.
#> 14 Germany                  Espirito Santo            1741.
#> 15 United Kingdom           Espirito Santo            1467.
#> 16 United States of America Espirito Santo            4464.
#> 17 Argentina                Espirito Santo           16125.
#> 18 Chile                    Espirito Santo            2375.
#> 19 Uruguay                  Espirito Santo            2073.
#> 20 Paraguay                 Espirito Santo            2340.
```

### Model

```r
set.seed(9000)
estimate_risk_spread(
  ## These arguments specified by user
  location_code = "Espirito Santo",
  r_incubation = function(n) rlnorm(n, 1.46, 0.35),
  r_infectious = function(n) rnorm(n, 4.5, 1.5/1.96),
  n_sim = 1e5,
  ## These arguments will be done inside the function
  location_population = get_vars(ef.es, "pop_size", id = FALSE)["Espirito Santo", ],
  num_cases_time_window = get_vars(ef.es, "num_cases_time_window", id = FALSE)["Espirito Santo", ],
  first_date_cases = get_vars(ef.es, "first_date_cases", id = FALSE)["Espirito Santo", ],
  last_date_cases = get_vars(ef.es, "last_date_cases", id = FALSE)["Espirito Santo", ],
  num_travellers_to_other_locations = get_n(ef.es, from = "Espirito Santo"),
  num_travellers_from_other_locations = get_n(ef.es, to   = "Espirito Santo"),
  avg_length_stay_days = na.omit(get_vars(ef.es, "duration_stay", id = FALSE, vector = TRUE))
)
#> Exportations done
#> Importations done
#>                          mean_cases lower_limit_95CI upper_limit_95CI
#> Italy                     0.2253813        0.1525319        0.3302201
#> Spain                     0.2278242        0.1541852        0.3337993
#> Portugal                  0.2319846        0.1570008        0.3398949
#> Germany                   0.1866430        0.1263153        0.2734631
#> United Kingdom            0.1771146        0.1198681        0.2595055
#> United States of America  0.9264530        0.6270100        1.3574295
#> Argentina                 1.1112801        0.7645684        1.4347439
#> Chile                     0.2609616        0.1794491        0.3479807
#> Uruguay                   0.2255244        0.1632340        0.2911472
#> Paraguay                  0.1548207        0.1216586        0.1838646
```

Hopefully, when I get the variables formatted and documented correctly (early next week), the model step will look like this:

```r
estimate_risk_spread(
  epiflows      = ef.es,
  location_code = "Espirito Santo",
  r_incubation  = function(n) rlnorm(n, 1.46, 0.35),
  r_infectious  = function(n) rnorm(n, 4.5, 1.5/1.96),
  n_sim         = 1e5
)
```

Created on 2018-07-20 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by zkamvar** [comment]
Update in the `new_version` branch. It works:

```r
set.seed(2018-07-25)
res <- estimate_risk_spread(Brazil_epiflows, 
                            location_code = "Espirito Santo",
                            r_incubation = function(n) rlnorm(n, 1.46, 0.35),
                            r_infectious = function(n) rnorm(n, 4.5, 1.5/1.96),
                            n_sim = 1e5
)
#> Exportations done
#> Importations done
res
##                          mean_cases lower_limit_95CI upper_limit_95CI
## Italy                     0.2233656        0.1520966        0.3078136
## Spain                     0.2255171        0.1537452        0.3126801
## Portugal                  0.2317019        0.1565528        0.3383112
## Germany                   0.1864162        0.1259548        0.2721890
## United Kingdom            0.1613418        0.1195261        0.2089475
## United States of America  0.9253419        0.6252207        1.3511047
## Argentina                 1.1283506        0.7623865        1.6475205
## Chile                     0.2648277        0.1789370        0.3866836
## Uruguay                   0.2408942        0.1627681        0.3517426
## Paraguay                  0.1619724        0.1213114        0.1926966
```

Also, it is possible for the user to override values in the data:

```r
set.seed(2018-07-25)
res <- estimate_risk_spread(Brazil_epiflows, 
                            location_code = "Espirito Santo",
                            r_incubation = function(n) rlnorm(n, 1.46, 0.35),
                            r_infectious = function(n) rnorm(n, 4.5, 1.5/1.96),
                            n_sim = 1e5,
                            location_population = 1e6
)
#> Exportations done
#> Importations done
res
#>                          mean_cases lower_limit_95CI upper_limit_95CI
#> Italy                     0.8875873        0.6043858         1.223158
#> Spain                     0.8961366        0.6109368         1.242496
#> Portugal                  0.9207131        0.6220933         1.344346
#> Germany                   0.7407616        0.5005064         1.081596
#> United Kingdom            0.6411233        0.4749603         0.830294
#> United States of America  3.6770282        2.4844376         5.368881
#> Argentina                 4.4837233        3.0294929         6.546747
#> Chile                     1.0523451        0.7110414         1.536564
#> Uruguay                   0.9572407        0.6467912         1.397719
#> Paraguay                  0.6436293        0.4820546         0.765718
```

**Comment by zkamvar** [comment]
This was closed in #19
